File: api/python/saveSVG.py
import json
import sys
import io

svg_data = sys.stdin.read()

def save_svg(svg_data, output_path):


    print(svg_data)

    with io.open(output_path, 'w', encoding='utf-8') as file:
        file.write(svg_data)

if __name__ == '__main__':

    
    if len(sys.argv) != 2:
        print("Usage: python save_svg.py 'svg_data' 'output_path'")
    else:
        output_path = sys.argv[1]

        save_svg(svg_data, output_path)

File: api/routes/saveSVG.js
var express = require("express");
var router = express.Router();
const { spawn } = require("child_process");

router.post("/", (req, res) => {
  var data = req.body;
  const svgImage = data.svgData;

  // spawn new child process to call the python script
  const pythonProcess = spawn("python", [
    "python/saveSVG.py",
    "./svg/result.svg",
  ]);

  pythonProcess.stdin.write(svgImage);
  pythonProcess.stdin.end();

  pythonProcess.stdout.on("data", (data) => {
    console.log(`Python script stdout: ${data}`);
  });

  pythonProcess.stderr.on("data", (data) => {
    console.error(`Python script stderr: ${data}`);
  });

  pythonProcess.on("close", (code) => {
    console.log(`Python script exited with code ${code}`);
    if (code === 0) {
      res.status(200).json({ result: "SVG saved" });
    } else {
      res.status(500).json({ message: "Error: SVG not saved" });
    }
  });
});

module.exports = router;
